src/babashka/neil/version.clj

```diff
@@ -0,0 +1,111 @@
+(ns babashka.neil.version
+  (:require [babashka.fs :as fs]
+            [babashka.neil.project :as proj]
+            [babashka.process :refer [sh]]
+            [clojure.edn :as edn]
+            [clojure.pprint :as pprint]
+            [clojure.string :as str]))
+
+(defn print-version-help []
+  (println (str/trim "
+Usage: neil version [major|minor|patch] [version]
+
+Bump the :version key in the project config.")))
+
+(defn current-version [deps-map]
+  (get-in deps-map [:aliases :neil :project :version]))
+
+(defn override-version [k {:keys [major minor patch]}]
+  (case k
+    :major {:major major :minor 0 :patch 0}
+    :minor {:major major :minor minor :patch 0}
+    :patch {:major major :minor minor :patch patch}))
+
+(defn bump-version [k opts]
+  (override-version k (update opts k inc)))
+
+(def zero-version {:major 0 :minor 0 :patch 0})
+
+(defn save-version-bump [deps-map sub-command override opts]
+  (let [prev-v (current-version deps-map)
+        next-v (if override
+                 (do
+                   (when (<= override (get prev-v sub-command))
+                     (throw (ex-info (format "Invalid %s number provided" (name sub-command))
+                                     {:current prev-v
+                                      :input {sub-command override}})))
+                   (merge zero-version
+                          (override-version sub-command (assoc prev-v sub-command override))))
+                 (bump-version sub-command (or prev-v zero-version)))]
+    (proj/assoc-project-meta! {:deps-file (:deps-file opts) :k :version :v next-v})
+    next-v))
+
+(def valid-version-keys
+  #{:major :minor :patch})
+
+(defn git-opts [opts]
+  (let [root (str (fs/parent (:deps-file opts)))]
+    (when (seq root)
+      {:dir root})))
+
+(defn git-repo? [opts]
+  (str/blank? (:err (sh "git status --porcelain" (git-opts opts)))))
+
+(defn git-clean-working-directory? [opts]
+  (let [{:keys [out err]} (sh "git status --porcelain" (git-opts opts))]
+    (and (str/blank? err) (str/blank? out))))
+
+(defn version-map->str [{:keys [major minor patch qualifier]}]
+  (str (str/join "." [major minor patch])
+       (when qualifier (str "-" qualifier))))
+
+(defn git-add [opts]
+  (sh "git add deps.edn"
+      (assoc (git-opts opts) :err :inherit)))
+
+(defn git-commit [version opts]
+  (sh ["git" "commit" "-m" (version-map->str version)]
```

Comment:
Maybe the message could have a `v` in front of it?

Reply:
I'm copying `npm`'s behavior here: the commit message is just `1.0.0` and the tag is `v1.0.0`. Would you prefer to add the `v` still or leave it out and match `npm`'s behavior? Either way seems fine to me.

Reply:
I think I prefer the v, it makes for a clearer commit message

Reply:
Sounds good.